src/components/quotes.js

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from 'react';
+
+function Quotes() {
+  const [quotes, setQuotes] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const category = 'computers';
+    const fetchData = async () => {
+      try {
+        const res = await fetch(`https://api.api-ninjas.com/v1/quotes?category=${category}`, {
+          headers: {
+            'X-Api-Key': 'CkL9zePipDK5tgJtxyNgcQ==uOajyB7tQnvSQGi0',
+          },
+        });
+        const data = await res.json();
+        setQuotes(data);
+      } catch (error) {
+        setQuotes([]);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    fetchData();
+  }, []);
+
+  if (isLoading) {
+    return <div className="loading">Loading quotes ...</div>;
+  }
+
+  if (!quotes.length) {
+    return (
+      <div className="failed">
+        Failed to load quotes.
+      </div>
+    );
```

Comment:
- [ ] You have done a great job👍 adding the `isLoading` state. 🙏Please consider adding another state called `hasError`, like so: ``` const [hasError, setHasError] = useState(false) ```. You can then set this state in a catch block like this:
```
Copy code
} catch (error) {
  setHasError(true)
}
```
This is considered best✔ practice and will make your code more robust since catch blocks are used to catch errors.